### ISSUE TITLE
SIGTERM is sent twice on Linux

### ISSUE THREAD

**Issue by Hornwitser** [clusterio/factorioClusterio#217]
Sending ^C via terminal on Linux sends SIGTERM to the factorio process twice, causing it to exit without completing the saving.  I.e:

```
$ node client.js start foo
...
^CFact:    2.822 Received SIGINT, shutting down   2.822 Quitting: signal.
   2.822 Info MainLoop.cpp:400: Saving map as instances/foo/saves/_autosave1.zip

Caught interrupt signal, disconnecting RCON
Rcon disconnected, Sending ^C
Fact:    2.825 Received second SIGINT, terminating immediately
child process exited with code 1
```

The reason for this is that SIGINT sent via the terminal is sent by default to all members of the active _process group_ (not to be confused by the permission group the process runs as, that's something else entirely.)  So both client.js and factorio is receiving the SIGINT, then client.js sends another SIGINT to the factorio process.

It's still possible that only the client.js process is sent SIGTERM, process managers may be doing this instead of sending to the process group.

Fixing this might be a simple as [setting the detached option](https://nodejs.org/docs/latest-v10.x/api/child_process.html#child_process_options_detached) to `spawn`, but it needs testing.

**Comment by Danielv123** [comment]
Reading the linked docs, that seems that it would make windows and linux behave in mostly the same way (let subprocess run by itself until explicitly ordered to shut down) instead of the current platform dependant mess.

**Comment by psihius** [comment]
Easy to test, easy PR. I'll see if I have time this evening to test and make a PR.

**Comment by Hornwitser** [comment]
And that's a no.  RCON doesn't recognise /quit.  I guess you're not supposed to manage Factorio servers on Windows then.

**Comment by Danielv123** [comment]
One option (also ugly) would be to wait to see if ctrl+c is detected in the logs, and if its not detected in a second or so send a second ctrl+c. That would work in every terminal and cross platform, but be quite ugly.

**Comment by Hornwitser** [comment]
You can't send a CTRL+C to a process in Windows.  It simply doesn't work that way.  CTRL+C is inherently tied to the terminal, and all processes attached to the terminal gets it (note that there is an API in windows to invoke it but whether or not node exposes a function to do this is another question.)

**Comment by Danielv123** [comment]
Does every process get it when stopping through a process manager as well? If not, we have to support both.

**Comment by Hornwitser** [comment]
There's no API in Windows to send CTRL+C to a process.  Whatever option a process manager provides, it cannot send a CTRL+C to just the node client.

**Comment by Hornwitser** [comment]
Fixed by #239
